tools/rtpsrelay/RelayThreadMonitor.h

```diff
@@ -0,0 +1,86 @@
+#ifndef RelayThreadMonitor_H_
+#define RelayThreadMonitor_H_
+
+#include <dds/DCPS/ThreadMonitor.h>
+#include <dds/DCPS/TimeTypes.h>
+#include <ace/Guard_T.h>
+#include <ace/Condition_Thread_Mutex.h>
+#include <ace/Log_Msg.h>
+#include <dds/DCPS/RcObject.h>
+
+#include <map>
+#include <deque>
+#include <string>
+#include <stack>
+
+namespace RtpsRelay {
+
+/**
+ * @class RelayThreadMonitor
+ *
+ * @brief Defines the means of tracking thread utilization by measuring
+ * time spent in event handling vs idle
+ */
+class RelayThreadMonitor : public OpenDDS::DCPS::ThreadMonitor {
```

Comment:
This should be read the builtin topic for status load.  It should cache the load for queries later.  It should detect deadlocks and abort if it detects a stuck thread.

Reply:
We will picked up later.